samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoAvailableNotificationServiceImpl.java

```diff
@@ -0,0 +1,369 @@
+                replyTo.append(displayName);
+                replyTo.append("' <");
+                replyTo.append(userEmail);
+                replyTo.append(">, ");
+            }
+        }
+        // Return instructors without trailing comma
+        return replyTo.length() > 0 ? replyTo.substring(0, replyTo.length() - 2) : "";
+    }
+
+    private String getUserEmail(String userId) {
+        String email = null;
+        try {
+            email = userDirectoryService.getUser(userId).getEmail();
+        } catch (UserNotDefinedException e) {
+            log.warn("Cannot get email for id: " + userId + " : " + e.getClass() + " : " + e.getMessage());
+        }
+        return email;
+    }
+
+    private String getUserDisplayName(String userId) {
+        String userDisplayName = "";
+        try {
+            userDisplayName = userDirectoryService.getUser(userId).getDisplayName();
+        } catch (Exception e) {
+            log.debug("Could not get user {}", userId + e);
+        }
+        return userDisplayName;
+    }
+
+    private String getUserFirstName(String userId) {
+        String email = "";
+        try {
+            email = userDirectoryService.getUser(userId).getFirstName();
+        } catch (UserNotDefinedException e) {
+            log.warn("Cannot get first name for id: " + userId + " : " + e.getClass() + " : " + e.getMessage());
+        }
+        return email;
+    }
+
+    private String getSetupRequest() {
+        return serverConfigurationService.getString("setup.request", "no-reply@" + serverConfigurationService.getServerName());
+    }
+
+    private String getServiceName() {
+        return serverConfigurationService.getString("ui.service", "Sakai");
+    }
+
+    private String getTimeLimit (Integer seconds){
+        StringBuilder output = new StringBuilder();
+        int seconds2 = seconds.intValue();
+        int hours = 0;
+        while (seconds2/60 > 59){
+            hours = hours + 1;
+            seconds2 = seconds2 - 3600;
+        }
+        if (hours > 0) {
+            output.append(hours + " hour(s), ");
+        }
+        output.append(seconds2/60 + " minute(s)");
+        return output.toString();
+    }
+
+    private Integer convertToSeconds(int hours, int minutes){
+        return (hours * 3600) + (minutes * 60);
+    }
+
+    private boolean ifSiteHasSamigo(Site site){
+        int countPages = 0;
+        log.debug("Execution has arrived at ifSiteHasSamigo.");
+        int countTools = 0;
+        while (countPages < site.getPages().size()){
+            while (countTools < site.getPages().get(countPages).getTools().size()){
+                if (StringUtils.equals(site.getPages().get(countPages).getTools().get(countTools).getToolId(), "sakai.samigo")){
+                    return true;
+                }
+                countTools = countTools + 1;
+            }
+            countTools = 0;
+            countPages = countPages + 1;
+        }
+        return false;
```

Comment:
If this method's purpose is solely to determine if the given `Site` has the Samigo tool, there is an easier way to do this:

```suggestion
        return site.getToolForCommonId("sakai.samigo") != null ? true : false;
```